src/vi/goctruyentranhvui/src/eu/kanade/tachiyomi/extension/vi/goctruyentranhvui/GocTruyenTranhVui.kt

```diff
@@ -0,0 +1,181 @@
+import keiyoushi.utils.parseAs
+import keiyoushi.utils.tryParse
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class GocTruyenTranhVui() : ParsedHttpSource() {
+    override val lang = "vi"
+
+    override val baseUrl = "https://goctruyentranhvui17.com"
+
+    override val name = "Goc Truyen Tranh Vui"
+
+    private val apiUrl = "$baseUrl/api/v2"
+
+    private val searchUrl = "$baseUrl/api/comic/search"
+
+    private val dateFormat = SimpleDateFormat("dd-MM-yyyy", Locale.US)
+
+    override val supportsLatest: Boolean = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val urls = client.newCall(GET(baseUrl + manga.url, headers)).execute().asJsoup()
+        val mangaId = checkChapterLists(urls)
+        return client.newCall(GET("$baseUrl/api/comic/$mangaId/chapter?offset=21&limit=-1", headers))
+            .asObservableSuccess()
+            .map { response ->
+                val res = response.parseAs<ChapterDTO>()
+                res.result.chapters.map { itm ->
+                    SChapter.create().apply {
+                        name = itm.numberChapter
+                        date_upload = dateFormat.tryParse(itm.stringUpdateTime)
+                        setUrlWithoutDomain("$baseUrl${manga.url}/chuong-${itm.numberChapter}")
+                    }
+                }
+            }
+    }
+
+    private fun checkChapterLists(document: Document) = document.selectFirst("input[id=comic-id-comment]")!!.attr("value")
+
+    override fun chapterFromElement(element: Element): SChapter = throw UnsupportedOperationException()
+
+    override fun chapterListSelector(): String = ""
+
+    override fun imageUrlParse(document: Document): String = ""
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        element.selectFirst("a.mt-1").let {
+            setUrlWithoutDomain(it!!.absUrl("href"))
+            title = it.text()
+        }
+        thumbnail_url = element.selectFirst("img.lazy")!!.absUrl("data-original")
+    }
+
+    override fun latestUpdatesNextPageSelector(): String = ".ma-3 a"
+
+    override fun latestUpdatesRequest(page: Int): Request = GET(
+        baseUrl.toHttpUrl().newBuilder().apply {
+            addPathSegments("truyen-cap-nhat")
+            addQueryParameter("p", page.toString())
+        }.build(),
+        headers,
+    )
```

Comment:
Do a search API call with "updated date" sort to get the same result.

Reply:
Then try the search API with updated date sort. You can find it on the search page.

Reply:
> Then try the search API with updated date sort. You can find it on the search page.

Always enable cloudflare except `truyen-cap-nhat` no enable cloudflare I not change it. But I will use `HttpSource()` 

<img width="865" height="67" alt="image" src="https://github.com/user-attachments/assets/579b1982-c494-4a7c-85a9-5241b8beb981" />